File: packages/room-server/src/database/entities/datasheet.record.alarm.entity.ts
/**
 * APITable <https://github.com/apitable/apitable>
 * Copyright (C) 2022 APITable Ltd. <https://apitable.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import { BaseEntity } from 'shared/entities/base.entity';
import { Column, Entity } from 'typeorm';

@Entity('datasheet_record_alarm')
export class DatasheetRecordAlarmEntity extends BaseEntity {
  @Column({
    name: 'alarm_id',
    nullable: false,
    unique: true,
    comment: 'alarm ID',
    length: 50,
  })
  alarmId!: string;

  @Column({
    name: 'space_id',
    nullable: false,
    comment: 'space ID',
    length: 50,
  })
  spaceId!: string;

  @Column({
    name: 'dst_id',
    nullable: false,
    comment: 'datasheet ID',
    length: 50,
  })
  dstId!: string;

  @Column({
    name: 'record_id',
    nullable: false,
    comment: 'datasheet record ID',
    length: 50,
  })
  recordId!: string;

  @Column({
    name: 'field_id',
    nullable: false,
    comment: 'datasheet field ID',
    length: 50,
  })
  fieldId!: string;

  @Column({
    name: 'alarm_at',
    nullable: false,
    comment: 'alarm time',
  })
  alarmAt!: Date;

  @Column({
    name: 'alarm_status',
    nullable: false,
    comment: 'alarm status(0-pending, 1-processing, 2-done, 3-failed)',
    width: 1,
    type: 'tinyint',
    default: () => 0,
  })
  status!: number;
}

File: backend-server/application/src/main/java/com/apitable/workspace/entity/DatasheetRecordAlarmEntity.java
/*
 * APITable <https://github.com/apitable/apitable>
 * Copyright (C) 2022 APITable Ltd. <https://apitable.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.apitable.workspace.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * Workbench - Datasheet Record Alarm Table.
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "datasheet_record_alarm")
public class DatasheetRecordAlarmEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Primary Key.
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * Alarm ID.
     */
    private String alarmId;

    /**
     * Space ID(link#space#space_id).
     */
    private String spaceId;

    /**
     * Datasheet ID(link#datasheet#dst_id).
     */
    private String dstId;

    /**
     * Record ID(link#datasheet_record#record_id).
     */
    private String recordId;

    /**
     * Field ID.
     */
    private String fieldId;

    /**
     * Alarm Time.
     */
    private LocalDateTime alarmAt;

    /**
     * Status: 0 - default, 1 - processing, 2 - process succeeded, 3 - process failed.
     */
    private Integer alarmStatus;

    /**
     * Delete Tag(0: No, 1: Yes).
     */
    @TableLogic
    private Integer isDeleted;

    /**
     * Create User.
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * Last Update By.
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * Create Time.
     */
    private LocalDateTime createdAt;

    /**
     * Update Time.
     */
    private LocalDateTime updatedAt;

}
